### PR TITLE
Update Changelog to changes made for v0.1.0

### PR DESCRIPTION
**Pull request by nathanshammah**
Description
-----------
Update Changelog to changes made for v0.1.0

If you have not finished all tasks in the checklist, you can also open a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/) to let the others know this on-going work and keep this checklist in the PR description.

Checklist
-----------
Please make sure you have finished the following tasks before requesting a review of the pull request (PR). For more information, please refer to the [Contributor's Guide](CONTRIBUTING.md):

- [ ] Contributions to mitiq should follow the [pep8 style](https://www.python.org/dev/peps/pep-0008/). You can enforce it easily with [`black`][black] style and with [`flake8`][flake8] conventions.
- [ ] Please add tests to cover your changes, if applicable, and make sure that all new and existing tests pass locally.
- [ ] If the behavior of the code has changed or new feature has been added, please also [update the documentation](docs/CONTRIBUTING_DOCS.md).
- [ ] Functions and classes have useful [Google-style][google] docstrings and [type hints][hints] in the signature of the objects.
- [ ] (Bug fix) The associated issue is referenced above using [auto-close keywords][auto-close]. If the PR fixes an issue, use the keyword fix/fixes/fixed followed by the issue id. For example, if the PR fixes issue 1184, type "Fixes #1184" (without quotes).
- [x] The [changelog][changelog] is updated, including author and PR number (@username, gh-xxx).

[auto-close]: https://help.github.com/en/articles/closing-issues-using-keywords
[black]: https://black.readthedocs.io/en/stable/index.html
[changelog]: https://github.com/unitaryfund/mitiq/blob/master/CHANGELOG.md
[contributing]: https://github.com/unitaryfund/mitiq/blob/master/CONTRIBUTING.md
[docs]: https://github.com/unitaryfund/mitiq/blob/master/docs/source/
[flake8]: http://flake8.pycqa.org
[google]: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html
[hints]: https://www.python.org/dev/peps/pep-0484/
[pep-484]: https://www.python.org/dev/peps/pep-0484/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,19 +1,65 @@
...
+- Run CI on all commits in PRs to master and close #316 (@karalekas, gh-#325).
+
+- Add Unitary Fund logo to the documentation html and close #297 (@nathanshammah, gh-#323).
+
+- Add circuit conversion error + tests (@rmlarose, gh-#321).
+
+- Make test file names unique (@rmlarose, gh-#319).
+
+- Update package version from v. 0.1a2, released, to 0.10dev (@nathanshammah, gh-#314).
+>>>>>>> master
```
Comment (rmlarose) [inline CHANGELOG.md:52]:
```suggestion
```

**Comment by willzeng** [comment]
I lean towards a higher level summary rather than going through every commit. The git history already has all the commits with authors and issues. The changelog is then a more easily human readable summary.

Am open to being persuaded otherwise of course. @nathanshammah

**Comment by nathanshammah** [comment]
I am fine either way. I just followed the prescriptions that are contained in the ``.github`` folder for a PR template, going through the merged ones. Let me know what to do as this is a blocker for the release.

**Comment by rmlarose** [comment]
I also prefer a higher level summary, I think that's easier for users to look through and see how their code is affected by a new release.

We can decide on a specific format later, it isn't too important now since this is blocking for the release.

**Comment by willzeng** [comment]
Sounds good. @nathanshammah can you reformat this into a summary of those changes? For major features you could include a link to the gh-issue but for others you can summarize.

After that then this should be good to merge. Excited about this release!

**Comment by karalekas** [comment]
@willzeng how would you feel about doing both? one of the really nice things about the per-commit summary is that you can have contributors add a line as part of a PR, and then it's trivial to write a summary upon release (rather than combing through closed PRs trying to figure out what was part of this or that release)

**Comment by willzeng** [comment]
Good point, Peter. How about we do have a per-PR summary get added to the changelog one by one, but then when we cut a release we do summarize and edit those into the shipped release changelog?

**Comment by karalekas** [comment]
Sure, if you think having both is too much content, then that works for me!

For clarity, this is an example of what I mean by "both":

# Changelog

## Version 0.1.0 (August 31st, 2020)

### Summary

This marks the first public release of mitiq...

### All Changes

- Add to the documentation instructions for maintainers to make a new release (@nathanshammah, gh-#332).
- Add basic compilation facilities, don't relabel qubits (@karalekas, gh-#324)
- Update readme (@rmlarose, gh-#330).
- Add mypy type checking to CI, resolve existing issues (@karalekas, gh-#326).
- Add readthedocs badge to readme (@nathanshammah, gh-#329).
- Add change log as markdown file (@nathanshammah, gh-#328).
- Add documentation on mitigating the energy landscape for QAOA MaxCut on two qubits (@rmlarose, gh-#241).
- Simplify inverse gates before conversion to QASM (@andreamari, gh-#283).
- Restructure library with ``zne/`` subpackage, modules renaming (@nathanshammah, gh-#298).
- [Bug Fix] Fix minor problems in executors documentation (@andreamari, gh-#292).
- Add better link to docs and more detailed features (@andreamari, gh-#306).
- [Bug Fix] Fix links and author list in README (@willzeng, gh-#302).
- Add new sections and more explanatory titles to the documentation's guide (@nathanshammah, gh-#285).
- Store optimal parameters after calling reduce in factories (@rmlarose, gh-#318).
- Run CI on all commits in PRs to master and close #316 (@karalekas, gh-#325).
- Add Unitary Fund logo to the documentation html and close #297 (@nathanshammah, gh-#323).
- Add circuit conversion error + tests (@rmlarose, gh-#321).
- Make test file names unique (@rmlarose, gh-#319).
- Update package version from v. 0.1a2, released, to 0.10dev (@nathanshammah, gh-#314).

**Comment by nathanshammah** [comment]
I like your proposal @karalekas. Thanks for clarifying this point on the indexing standard. I will update the PR.

**Comment by karalekas** [comment]
@nathanshammah just one small nitpick -- could you get rid of the extra newlines between the bullet points in the changelog? it's much more compact that way

e.g.


- the truth

- is out there

vs 

- the truth
- is out there

**Pull request merged**